addons/mail/static/src/models/message_reaction_per_emoji/message_reaction_per_emoji.js

```diff
@@ -0,0 +1,79 @@
+/** @odoo-module **/
+
+import { registerNewModel } from '@mail/model/model_core';
+import { attr, many2one, one2many} from '@mail/model/model_field';
+import { clear, insert, insertAndReplace, link, replace, unlink, unlinkAll } from '@mail/model/model_field_command';
+
+function factory(dependencies) {
+
+    class MessageReactionPerEmoji extends dependencies['mail.model'] {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         *  Removes the reaction for current user by clicking the message emoji reaction.
+         */
+        removeReaction() {
+            this.message.onEmojiSelection(this.emoji_unicode);
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        static _createRecordLocalId(data) {
+            return `${this.modelName}_${data.messageId}_${data.emoji_unicode}`;
+        }
+
+        /**
+         * @private
+         */
+        _computeHasUserReacted() {
+            const reactionForCurrentUser = this.messaging.models['mail.message_reaction'].all().filter(r =>
+                    r.message.id === this.messageId && r.emoji_unicode === this.emoji_unicode && r.reactor === this.messaging.currentPartner)[0];
+            if (reactionForCurrentUser) {
+                return true;
+            }
+            return false;
+        }
+
+        /**
+         * @private
+         */
+        _computeMessage() {
+            return insert({ id: this.messageId });
+        }
+    }
+
+    MessageReactionPerEmoji.fields = {
+        count: attr({
+            required: true,
+        }),
+        emoji_unicode: attr({
+            required: true,
+        }),
```

Comment:
No snake case, and also don't enforce modelling of message reaction with emoji (and unicode representation in their content). Just defining field as the content of the reaction is good.